lib/manager/gitlabci/extract.js

```diff
@@ -0,0 +1,91 @@
+const { splitImageParts } = require('../docker/extract');
+
+module.exports = {
+  extractDependencies,
+};
+
+function extractDependencies(content) {
```

Comment:
Is there any reason not to use a yaml parser?

Reply:
The problem parsing causes is then how to maintain the exact formatting/whitespace once we update it. e.g. if we parse files, then update the internal structure, then write them back out, then there's a high chance that in many case we'll end up with "diffs" all over the file instead of just where we want them.

Therefore I've found it's better so far to do a "hacky" line-by-line parser in most cases, assuming we don't really need to understand/care about the internal structure anyway.